aider/sendchat.py

```diff
@@ -41,6 +42,14 @@ def send_with_retries(client, model_name, messages, functions, stream):
     if functions is not None:
         kwargs["functions"] = functions
 
+
+    # Check conditions to switch to gpt-4-vision-preview or strip out image_url messages
+    if client and is_gpt4_with_openai_base_url(model_name, client):
+        if any(isinstance(msg.get("content"), list) and any("image_url" in item for item in msg.get("content") if isinstance(item, dict)) for msg in messages):
+            kwargs['model'] = "gpt-4-vision-preview"
+            # gpt-4-vision is limited to max tokens of 4096
+            kwargs["max_tokens"] = 4096
```

Comment:
Do we need to add `max_tokens`? We don't set it for any other model right?

Reply:
@paul-gauthier 
Max tokens is necessary for gpt-4-vision